sharepoint-adfs/dsc/ConfigureSPVM.ps1

```diff
@@ -0,0 +1,757 @@
+            UserName = $SPSetupCreds.UserName
+            Password = $SPSetupCreds
+            PasswordNeverExpires = $true
+            Ensure = "Present"
+            DependsOn = "[xComputer]DomainJoin"
+        }
+
+        Group AddSPSetupAccountToAdminGroup
+        {
+            GroupName='Administrators'   
+            Ensure= 'Present'             
+            MembersToInclude= $SPSetupCredsQualified.UserName
+            Credential = $DomainAdminCredsQualified    
+            PsDscRunAsCredential = $DomainAdminCredsQualified
+            DependsOn = "[xADUser]CreateSPSetupAccount"
+        }
+
+        xADUser CreateSParmAccount
+        {
+            DomainAdministratorCredential = $DomainAdminCredsQualified
+            DomainName = $DomainFQDN
+            UserName = $SPFarmCreds.UserName
+            Password = $SPFarmCreds
+            PasswordNeverExpires = $true
+            Ensure = "Present"
+            DependsOn = "[xComputer]DomainJoin"
+        }
+
+        xADUser CreateSPSvcAccount
+        {
+            DomainAdministratorCredential = $DomainAdminCredsQualified
+            DomainName = $DomainFQDN
+            UserName = $SPSvcCreds.UserName
+            Password = $SPSvcCreds
+            PasswordNeverExpires = $true
+            Ensure = "Present"
+            DependsOn = "[xComputer]DomainJoin"
+        }
+
+        xADUser CreateSPAppPoolAccount
+        {
+            DomainAdministratorCredential = $DomainAdminCredsQualified
+            DomainName = $DomainFQDN
+            UserName = $SPAppPoolCreds.UserName
+            Password = $SPAppPoolCreds
+            PasswordNeverExpires = $true
+            Ensure = "Present"
+            DependsOn = "[xComputer]DomainJoin"
+        }
+
+        File AccountsProvisioned
+        {
+            DestinationPath = "F:\Logs\DSC1.txt"
+            PsDscRunAsCredential = $SPSetupCredential
+            Contents = "AccountsProvisioned"
+            Type = 'File'
+            Force = $true
+            DependsOn = "[Group]AddSPSetupAccountToAdminGroup", "[xADUser]CreateSParmAccount", "[xADUser]CreateSPSvcAccount", "[xADUser]CreateSPAppPoolAccount"
+        }
+
+        
+        #**********************************************************
+        # Download binaries and install SharePoint CU
+        #**********************************************************
+        File CopyCertificatesFromDC
+        {
+            Ensure = "Present"
+            Type = "Directory"
+            Recurse = $true
+            SourcePath = "\\$DCName\F$\Setup"
+            DestinationPath = "F:\Setup\Certificates"
+            Credential = $DomainAdminCredsQualified
+            DependsOn = "[File]AccountsProvisioned"
+        }
+
+        xRemoteFile DownloadLdapcp
+        {  
+            Uri             = $LdapcpLink
+            DestinationPath = "F:\Setup\LDAPCP.wsp"
+            DependsOn = "[File]AccountsProvisioned"
+        }        
+
+        <#
```

Comment:
on this file I removed some but if possible I prefer to leave the others as they are work in progress